ios/brave-ios/Sources/BraveStore/Subscription/SDK/BraveSkusSDK.swift

```diff
@@ -74,6 +75,39 @@ public struct SkusOrder: Codable {
   }
 }
 
+/// An enum representing the Skus SDK Environment
+public enum BraveSkusEnvironment {
+  case beta
+  case nightly
+  case release
+
+  /// Returns the current Skus SDK Environment
+  /// When DEBUG is defined, uses the Bundle-ID to determine the environment
+  /// Otherwise uses Brave-Core's version info.
+  public static var current: BraveSkusEnvironment {
+    if AppConstants.buildChannel == .release {
+      return .release
+    }
+
+    #if DEBUG
+    switch Bundle.main.bundleIdentifier {
+    case "com.brave.ios.browser.beta": return .beta
+    case "com.brave.ios.BrowserBeta": return .nightly
+    case "com.brave.ios.browser": return .release
+    default:
+      fatalError("[BraveSkusSDK] - BraveSkusEnvironment - Unknown Bundle-ID")
+    }
+    #else
+    switch BraveCoreVersionInfo.channel {
+    case .beta: return .beta
+    case .development, .nightly: return .nightly
+    case .stable, .unknown: fallthrough
+    default: return .release
+    }
+    #endif
```

Comment:
This is confusing as it each bit of logic is checking different things: 

- `AppConstants.buildChannel` is going to be based on the scheme you pick in Xcode, for instance `Debug (App Store)` is still going to be `AppBuildChannel.debug`.
- Bundle identifier is going to be based different based on xcconfig per scheme (e.g. `Debug` vs `Debug (App Store)`)
- BraveCore channel is going to be simply based on whether or not you passed a `--channel` into the brave-core build, which likely wont exist for local builds and it'll default to release anyways.

Reply:
Yes. Correct. That's exactly what I want.
So when testing Debug (No Core), I want it to return `nightly` because the bundle-id is nightly and payments only work based on the bundle-id.
When testing `Debug (AppStore)`, I want it to test against `release`/`stable` because the bundle-id is the same as the app-store release bundle-id.

But once a TF build is made, it must use the Brave-Core channel because I want QA to be able to use the switches for the environment.